python/hsfs/training_dataset.py

```diff
@@ -62,6 +62,9 @@ def __init__(
         from_query=None,
         querydto=None,
         label=None,
+        prepared_statement_connection=None,
+        prepared_statements=None,
+        serving_keys=None,
```

Comment:
Why do we need these? I don't see that you changed the DTO in hopsworks

Reply:
I am trying to find a good way to let users what are the primary key names they need to populate. `serving_keys` will return list of them. 

`prepared_statement_connection` and `prepared_statements` will cache related objects once they are initialized

Reply:
the serving keys are the primary keys and you already have them in the features. And you don't need them in the init function as they are only populated explicitly when calling the `get_vector` method.

Reply:
ahh you mean to create helper function that will return list of serving keys from feature primary keys, right?